### PR TITLE
Add numberOfLines prop to DataTableTitle

### LINKED ISSUE #848: DataTable Header Text Wrap
### Current behaviour

Unable to wrap the header title
screenshot 2019-02-18 at 12 32 44 pm

### Expected behaviour
I would like to add some style or atleast try to wrap the header title like the given image below

screenshot 2019-02-18 at 12 35 20 pm

### Your Environment

| software              | version
| --------------------- | -------
| ios         | ios 12
| react-native          | 0.57.7
| react-native-paper   | 2.11

Any help is much appreciated.

### PR DESCRIPTION
**Pull request by Taym95**
Add a `numberOfLines` prop to `DataTable.Title` 
Try to fix #848

### CODE REVIEW

File: src/components/DataTable/DataTableTitle.js [hunk 1]
```diff
@@ -99,7 +104,7 @@ class DataTableTitle extends React.Component<Props, State> {
               styles.cell,
               sortDirection ? styles.sorted : { color: textColor },
             ]}
-            numberOfLines={1}
+            numberOfLines={numberOfLines || 1}
```
Comment (satya164) [inline src/components/DataTable/DataTableTitle.js]: Use `defaultProps` for the default value

**Comment by satya164** [comment]
@allcontributors[bot] let's add @Taym95 to contributors for code

**Comment by iamautomaton** [comment]
Hi, i am developing the web app using react native. I found this plugin and it is useful for me. However, my datatable title is too long so i add numberofLines want to solve the problem, it doesn't work. It still showing "..." of my long title. Hope can fix this issue. Thanks

**Comment by Trancever** [comment]
@iamautomaton That's how `numberOfLines` works. It truncates the text and append 3 dots to the end.

**Comment by iamautomaton** [comment]
@Trancever Maybe i am not explaining clearly, i will make it more easy to understand.

I am writing a report page, i am designing a form which is showing the detail of some applicant information. i want to show "English Name of Applicant" on one of my form title of my report page. However, i set the numberOfLines to two, it still showing one line with "...". I am expecting it will show two line of the title 


Expected
> English Name of 
       Application

Actually
> English Name of App...

Below is my code and the ScreenCap

Code:

> <DataTable.Header>
....
<DataTable.Title numberOfLines={2} style={{alignSelf:'stretch',justifyContent:"center",padding:10,width:150,flexDirection:'row'}}><Text style={{flex: 1, flexWrap: 'wrap'}}>English Name of Applicate</Text></DataTable.Title>
<DataTable.Title style={{alignSelf:'stretch',justifyContent:"center",padding:10,width:150,flexDirection:'row' }}><Text style={{flex: 1, flexWrap: 'wrap'}}>Phone Number</Text></DataTable.Title>
<DataTable.Title style={{alignSelf:'stretch',justifyContent:"center",padding:10,width:150,flexDirection:'row' }}><Text style={{flex: 1, flexWrap: 'wrap'}}>Address</Text></DataTable.Title>
....
</DataTable.Header>

ScreenCap:
Simulator Screen Shot - iPad 5th 12 4 - 2019-12-12 at 09 21 06

**Comment by Surangaup** [comment]
> @Trancever Maybe i am not explaining clearly, i will make it more easy to understand.
> 
> I am writing a report page, i am designing a form which is showing the detail of some applicant information. i want to show "English Name of Applicant" on one of my form title of my report page. However, i set the numberOfLines to two, it still showing one line with "...". I am expecting it will show two line of the title
> 
> Expected
> 
> > English Name of
> > Applicant
> 
> Actually
> 
> > English Name of App...
> 
> Below is my code and the ScreenCap
> 
> Code:
> 
> > <DataTable.Header>
> > ....
> > <DataTable.Title numberOfLines={2} style={{alignSelf:'stretch',justifyContent:"center",padding:10,width:150,flexDirection:'row'}}><Text style={{flex: 1, flexWrap: 'wrap'}}>English Name of Applicant</DataTable.Title>
> > <DataTable.Title style={{alignSelf:'stretch',justifyContent:"center",padding:10,width:150,flexDirection:'row' }}><Text style={{flex: 1, flexWrap: 'wrap'}}>Phone Number</DataTable.Title>
> > <DataTable.Title style={{alignSelf:'stretch',justifyContent:"center",padding:10,width:150,flexDirection:'row' }}><Text style={{flex: 1, flexWrap: 'wrap'}}>Address</DataTable.Title>
> > ....
> > </DataTable.Header>
> 
> ScreenCap:
> Simulator Screen Shot - iPad 5th 12 4 - 2019-12-12 at 09 21 06

Did you fix this?

**Comment by mikehardy** [comment]
I think I repaired the original implementation so that it handles text component's special text alignment issues with wrapping, while allowing a cell height that is a multiple of numberOfLines - patch-package patch available here for testing: https://github.com/callstack/react-native-paper/pull/3015#issuecomment-996956251

**Comment by mikehardy** [comment]
My PR attempting to address this is release now in 4.12.0, hopefully it works for everyone!

**Pull request merged**